### PR TITLE
Debugging #857

### PR DESCRIPTION
**Pull request by 00vareladavid**
I rolled back some inessential changes to see if the error persists.

### CODE REVIEW

**Comment by 00vareladavid** [comment]
bors try

**Comment by 00vareladavid** [comment]
bors try

**Comment by 00vareladavid** [comment]
bors try

**Comment by 00vareladavid** [comment]
bors try

**Comment by 00vareladavid** [comment]
bors try

**Comment by 00vareladavid** [comment]
:triumph:

**Comment by 00vareladavid** [comment]
Well, I'm tempted to just leave it at this. This retains most of the benefits of #857, except that Project is represented by a Dict and not a struct.

Thoughts?

File: src/Operations.jl [hunk 1]
```diff
@@ -1159,21 +1156,22 @@ function rm(ctx::Context, pkgs::Vector{PackageSpec})
...
-        end
-        if isempty(ctx.env.project["targets"])
-            delete!(ctx.env.project, "targets")
-        end
+    filter!(ctx.env.project["targets"]) do (target, deps)
+        !isempty(filter!(in(deps_names), deps))
+    end
+    #= WHY delete?
+    if isempty(ctx.env.project["targets"])
+        delete!(ctx.env.project, "targets")
```
Comment (fredrikekre) [inline src/Operations.jl]: Oh, is this #876?

Comment (fredrikekre) [reply src/Operations.jl]: Although I don't know why it would be empty.

Comment (00vareladavid) [reply src/Operations.jl]: I don't think so; the "targets" section wouldn't be empty in that case, right?

Comment (KristofferC) [reply src/Operations.jl]: I think this can just be deleted.

Comment (KristofferC) [reply src/Operations.jl]: I think I was thinking you could `rm` a Test dependency and when the target section got empty we should remove the section.

Comment (fredrikekre) [reply src/Operations.jl]: Maybe something accidentally removed it from targets, and then this line removed the targets completely.

**Comment by StefanKarpinski** [comment]
I rather liked having the project as a struct. It means that you know exactly what the meaningful data in a project file is and the type system guarantees that it will have the expected types. I know one can always verify it first and then just assume that the types are correct later but it feels better to have the type of the project object itself encode this knowledge instead of just having it be implicit.

**Pull request closed without merging**